File: 프로그래머스/lv2/42586. 기능개발/기능개발.py
from math import ceil


def solution(progresses, speeds):
    answer = []
    days_left = [ceil((100-p)/s) for p, s in zip(progresses, speeds)]
    # 각 작업이 끝나기까지 남은 일수 계산

    while days_left:
        done = 1
        left = days_left.pop(0)
        # 첫 번째 작업은 먼저 배포

        while days_left and days_left[0] <= left:
            done += 1
            days_left.pop(0)
            # 같이 배포할 수 있는 기능들을 묶음

        answer.append(done)

    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <bits/stdc++.h>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    
    queue<int> Queue;
    
    for(const int& progress : progresses){
        Queue.push(progress);
    }
    
    int days = 0;
    int idx = 0;
    
    int cnt = 0;
    while(!Queue.empty()){
        int now = Queue.front();
        Queue.pop();
        
        int needDays = ceil(float(100 - progresses[idx]) / speeds[idx]);
        idx++;
        
        if(days >= needDays){
            cnt++;
        }
        else{
            if(cnt > 0){
                answer.push_back(cnt);
            }
            cnt = 1;
            days = needDays;
        }
    }
    
    if (cnt > 0) {
        answer.push_back(cnt);
    }
    
    return answer;
}